### PR TITLE
add some extra logging to dpop validation

### PR DESCRIPTION
**Pull request by brockallen**

### CODE REVIEW

**Review by leastprivilege — APPROVED** [review]

File: src/IdentityServer/Validation/Default/DefaultDPoPProofValidator.cs [hunk 1]
```diff
@@ -305,7 +305,7 @@ protected virtual async Task ValidatePayloadAsync(DPoPProofValidatonContext cont
         await ValidateReplayAsync(context, result);
         if (result.IsError)
         {
-            Logger.LogDebug("Detected replay of DPoP token");
+            result.ErrorDescription = "Detected replay of DPoP proof token.";
```
Comment (josephdecock) [inline src/IdentityServer/Validation/Default/DefaultDPoPProofValidator.cs:308]: Does this get sent back to the attacker? If so, why?

Comment (brockallen) [reply src/IdentityServer/Validation/Default/DefaultDPoPProofValidator.cs:308]: Yes -- but do you feel this info is a problem?

Comment (josephdecock) [reply src/IdentityServer/Validation/Default/DefaultDPoPProofValidator.cs:308]: Well, it leaks information to the attacker that this DPOP implementation is tracking tokens for replay. That seems minor, but I guess that might prompt an attacker to try to perform some sort of memory exhaustion attack. That brings up a separate issue, which is that we should be defending against memory exhaustion by rejecting jtis that are unreasonably long.

Comment (brockallen) [reply src/IdentityServer/Validation/Default/DefaultDPoPProofValidator.cs:308]: > Well, it leaks information to the attacker that this DPOP implementation is tracking tokens for replay.

Well, the code is on the internet :)

> which is that we should be defending against memory exhaustion by rejecting jtis that are unreasonably long

Good point. I thought we had done a size check somewhere. I'll look into this.

**Pull request merged**